packages/zip-it-and-ship-it/src/runtimes/node/in_source_config/index.ts

```diff
@@ -18,26 +19,42 @@ import { parse as parseSchedule } from './properties/schedule.js'
 
 export const IN_SOURCE_CONFIG_MODULE = '@netlify/functions'
 
-export type ISCValues = {
-  routes?: Route[]
-  schedule?: string
-  methods?: string[]
-  trafficRules?: TrafficRules
-  name?: string
-  generator?: string
-  timeout?: number
-}
-
 export interface StaticAnalysisResult extends ISCValues {
   inputModuleFormat?: ModuleFormat
   invocationMode?: InvocationMode
+  routes?: Route[]
   runtimeAPIVersion?: number
 }
 
 interface FindISCDeclarationsOptions {
   functionName: string
 }
 
+const httpMethods = z.preprocess(
+  (input) => (typeof input === 'string' ? input.toUpperCase() : input),
+  z.enum(['GET', 'POST', 'PUT', 'PATCH', 'OPTIONS', 'DELETE', 'HEAD']),
```

Comment:
Are you sure this list is exhaustive? What about uncommon methods like `TRACE` or `PROPFIND`? Might be worth doing this change in a separate PR and explore that first.

Reply:
These are the methods that our CDN supports, everything else is blocked: 
```
var validMethods = map[string]struct{}{
	"ACL":               {},
	"BASELINE-CONTROL":  {},
	"BIND":              {},
	"CHECKIN":           {},
	"CHECKOUT":          {},
	"COPY":              {},
	"DELETE":            {},
	"GET":               {},
	"HEAD":              {},
	"LABEL":             {},
	"LINK":              {},
	"LOCK":              {},
	"MERGE":             {},
	"MKACTIVITY":        {},
	"MKCALENDAR":        {},
	"MKCOL":             {},
	"MKREDIRECTREF":     {},
	"MKWORKSPACE":       {},
	"MOVE":              {},
	"OPTIONS":           {},
	"ORDERPATCH":        {},
	"PATCH":             {},
	"POST":              {},
	"PRI":               {},
	"PROPFIND":          {},
	"PROPPATCH":         {},
	"PUT":               {},
	"REBIND":            {},
	"REPORT":            {},
	"SEARCH":            {},
	"UNBIND":            {},
	"UNCHECKOUT":        {},
	"UNLINK":            {},
	"UNLOCK":            {},
	"UPDATE":            {},
	"UPDATEREDIRECTREF": {},
	"VERSION-CONTROL":   {},
	"*":                 {},
}
```